### PR TITLE
[FLINK-15385][table][docs] Translate SQL section of Table API into Chinese

### PR DESCRIPTION
**Pull request by ousheobin**
## What is the purpose of the change

Document Translation.

## Brief change log
- Translated the SQL section of Table API documents into Chinese

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/table/sql/queries.zh.md [hunk 1]
```diff
@@ -651,20 +651,20 @@ FROM
...
-        <p><a href="{{ site.baseurl }}/dev/table/streaming/temporal_tables.html">Temporal Tables</a> are tables that track changes over time.
-        A <a href="{{ site.baseurl }}/dev/table/streaming/temporal_tables.html#temporal-table">Temporal Table</a> provides access to the versions of a temporal table at a specific point in time.</p>
+        <p><a href="{{ site.baseurl }}/zh/dev/table/streaming/temporal_tables.html">Temporal Tables</a> 是随时间变化而变化的表。
+        <a href="{{ site.baseurl }}/zh/dev/table/streaming/temporal_tables.html#temporal-table">Temporal Table</a> 提供访问指定时间点的 temporal table 版本的功能。</p>
 
-        <p>Only inner and left joins with processing-time temporal tables are supported.</p>
-        <p>The following example assumes that <strong>LatestRates</strong> is a <a href="{{ site.baseurl }}/dev/table/streaming/temporal_tables.html#temporal-table">Temporal Table</a> which is materialized with the latest rate.</p>
+        <p>仅支持带有处理时间的 temporal tables 的 inner 和 left join。</p>
+        <p>下述示例中，假设 <strong>LatestRates</strong> 是一个根据最新的 rates 实现的 <a href="{{ site.baseurl }}/zh/dev/table/streaming/temporal_tables.html#temporal-table">Temporal Table</a> 。</p>
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
        <p>下述示例中，假设 <strong>LatestRates</strong> 是一个根据最新的 rates 物化的 <a href="{{ site.baseurl }}/zh/dev/table/streaming/temporal_tables.html#temporal-table">Temporal Table</a> 。</p>
```

File: docs/dev/table/sql/queries.zh.md [hunk 2]
```diff
@@ -863,33 +861,33 @@ FROM (
...
+**参数说明：**
 
-- `ROW_NUMBER()`: Assigns an unique, sequential number to each row, starting with one, according to the ordering of rows within the partition. Currently, we only support `ROW_NUMBER` as the over window function. In the future, we will support `RANK()` and `DENSE_RANK()`.
-- `PARTITION BY col1[, col2...]`: Specifies the partition columns. Each partition will have a Top-N result.
-- `ORDER BY col1 [asc|desc][, col2 [asc|desc]...]`: Specifies the ordering columns. The ordering directions can be different on different columns.
-- `WHERE rownum <= N`: The `rownum <= N` is required for Flink to recognize this query is a Top-N query. The N represents the N smallest or largest records will be retained.
-- `[AND conditions]`: It is free to add other conditions in the where clause, but the other conditions can only be combined with `rownum <= N` using `AND` conjunction.
+- `ROW_NUMBER()`: 根据当前分区内的各行的顺序从第一行开始，依次为每一行分配一个唯一且连续的号码。目前，我们只支持 `ROW_NUMBER` 在 over 窗口函数中使用。未来将会支持 `RANK()` 和 `DENSE_RANK()`函数。
+- `PARTITION BY col1[, col2...]`: 指定分区列，每个分区都将会有一个 Top-N 结果。
+- `ORDER BY col1 [asc|desc][, col2 [asc|desc]...]`: 指定排序列，不同列的排序方式可以不一样。
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
- `ORDER BY col1 [asc|desc][, col2 [asc|desc]...]`: 指定排序列，不同列的排序方向可以不一样。
```

File: docs/dev/table/sql/queries.zh.md [hunk 3]
```diff
@@ -925,26 +923,26 @@ val result1 = tableEnv.sqlQuery(
 </div>
 </div>
 
-#### No Ranking Output Optimization
+#### 不进行排序输出的优化
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
#### 无排名输出优化
```

File: docs/dev/table/sql/queries.zh.md [hunk 4]
```diff
@@ -925,26 +923,26 @@ val result1 = tableEnv.sqlQuery(
...
 
-As described above, the `rownum` field will be written into the result table as one field of the unique key, which may lead to a lot of records being written to the result table. For example, when the record (say `product-1001`) of ranking 9 is updated and its rank is upgraded to 1, all the records from ranking 1 ~ 9 will be output to the result table as update messages. If the result table receives too many data, it will become the bottleneck of the SQL job.
+如上文所描述，`rownum` 字段会作为唯一键的其中一个字段写到结果表里面，这会导致大量的结构写出到结果表。比如，当原始结果（名为 `product-1001` ）从排序第九变化为排序第一时，排名 1-9 的所有结果都会以更新消息的形式发送到结果表。若结果表收到太多的数据，将会成为 SQL 任务的瓶颈。
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
如上文所描述，`rownum` 字段会作为唯一键的其中一个字段写到结果表里面，这会导致大量的结果写出到结果表。比如，当原始结果（名为 `product-1001` ）从排序第九变化为排序第一时，排名 1-9 的所有结果都会以更新消息的形式发送到结果表。若结果表收到太多的数据，将会成为 SQL 任务的瓶颈。
```

File: docs/dev/table/sql/queries.zh.md [hunk 5]
```diff
@@ -1236,25 +1231,25 @@ val result4 = tableEnv.sqlQuery(
 
 {% top %}
 
-### Pattern Recognition
+## 模式匹配
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
### 模式匹配
```

File: docs/dev/table/sql/queries.zh.md [hunk 6]
```diff
@@ -1063,49 +1058,49 @@ val result1 = tableEnv.sqlQuery(
...
+      <td>定义一个滚动窗口。滚动窗口把行分配到有固定持续时间（ <code>interval</code> ）的不重叠的连续窗口。比如，5 分钟的滚动窗口以 5 分钟为间隔对行进行分组。滚动窗口可以定义在事件时间（批处理、流处理）或处理时间（流处理）上。</td>
     </tr>
     <tr>
       <td><code>HOP(time_attr, interval, interval)</code></td>
-      <td>Defines a hopping time window (called sliding window in the Table API). A hopping time window has a fixed duration (second <code>interval</code> parameter) and hops by a specified hop interval (first <code>interval</code> parameter). If the hop interval is smaller than the window size, hopping windows are overlapping. Thus, rows can be assigned to multiple windows. For example, a hopping window of 15 minutes size and 5 minute hop interval assigns each row to 3 different windows of 15 minute size, which are evaluated in an interval of 5 minutes. Hopping windows can be defined on event-time (stream + batch) or processing-time (stream).</td>
+      <td>定义一个跳跃的时间窗口（在 Table API 中称为滑动窗口）。滑动窗口有一个固定的持续时间（ 第二个 <code>interval</code> 参数 ）以及一个滑动的间隔（第一个 code>interval</code> 参数 ）。若滑动间隔小于窗口的持续时间，滑动窗口则会出现重叠；因此，行将会被分配到多个窗口中。比如，一个大小为 15 分组的滑动窗口，其滑动间隔为 5 分钟，将会把每一行数据分配到 3 个 15 分钟的窗口中。滑动窗口可以定义在事件时间（批处理、流处理）或处理时间（流处理）上。</td>
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
      <td>定义一个跳跃的时间窗口（在 Table API 中称为滑动窗口）。滑动窗口有一个固定的持续时间（ 第二个 <code>interval</code> 参数 ）以及一个滑动的间隔（第一个 <code>interval</code> 参数 ）。若滑动间隔小于窗口的持续时间，滑动窗口则会出现重叠；因此，行将会被分配到多个窗口中。比如，一个大小为 15 分组的滑动窗口，其滑动间隔为 5 分钟，将会把每一行数据分配到 3 个 15 分钟的窗口中。滑动窗口可以定义在事件时间（批处理、流处理）或处理时间（流处理）上。</td>
```

File: docs/dev/table/sql/queries.zh.md [hunk 7]
```diff
@@ -1116,68 +1111,68 @@ The start and end timestamps of group windows as well as time attributes can be
         <code>HOP_START(time_attr, interval, interval)</code><br/>
         <code>SESSION_START(time_attr, interval)</code><br/>
       </td>
-      <td><p>Returns the timestamp of the inclusive lower bound of the corresponding tumbling, hopping, or session window.</p></td>
+      <td><p>返回相对应的滚动，滑动和会话窗口范围内的下界时间戳。</p></td>
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
      <td><p>返回相对应的滚动、滑动和会话窗口范围内的下界时间戳。</p></td>
```

File: docs/dev/table/sql/queries.zh.md [hunk 8]
```diff
@@ -1116,68 +1111,68 @@ The start and end timestamps of group windows as well as time attributes can be
...
     </tr>
     <tr>
       <td>
         <code>TUMBLE_END(time_attr, interval)</code><br/>
         <code>HOP_END(time_attr, interval, interval)</code><br/>
         <code>SESSION_END(time_attr, interval)</code><br/>
       </td>
-      <td><p>Returns the timestamp of the <i>exclusive</i> upper bound of the corresponding tumbling, hopping, or session window.</p>
-        <p><b>Note:</b> The exclusive upper bound timestamp <i>cannot</i> be used as a <a href="{{ site.baseurl }}/dev/table/streaming/time_attributes.html">rowtime attribute</a> in subsequent time-based operations, such as <a href="#joins">time-windowed joins</a> and <a href="#aggregations">group window or over window aggregations</a>.</p></td>
+      <td><p>返回相对应的滚动，滑动和会话窗口<i>范围以外</i>的上界时间戳。</p>
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
      <td><p>返回相对应的滚动、滑动和会话窗口<i>范围以外</i>的上界时间戳。</p>
```

File: docs/dev/table/sql/queries.zh.md [hunk 9]
```diff
@@ -1116,68 +1111,68 @@ The start and end timestamps of group windows as well as time attributes can be
...
     </tr>
     <tr>
       <td>
         <code>TUMBLE_ROWTIME(time_attr, interval)</code><br/>
         <code>HOP_ROWTIME(time_attr, interval, interval)</code><br/>
         <code>SESSION_ROWTIME(time_attr, interval)</code><br/>
       </td>
-      <td><p>Returns the timestamp of the <i>inclusive</i> upper bound of the corresponding tumbling, hopping, or session window.</p>
-      <p>The resulting attribute is a <a href="{{ site.baseurl }}/dev/table/streaming/time_attributes.html">rowtime attribute</a> that can be used in subsequent time-based operations such as <a href="#joins">time-windowed joins</a> and <a href="#aggregations">group window or over window aggregations</a>.</p></td>
+      <td><p>返回相对应的滚动，滑动和会话窗口<i>范围以内</i>的上界时间戳。</p>
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
      <td><p>返回相对应的滚动、滑动和会话窗口<i>范围以内</i>的上界时间戳。</p>
```

File: docs/dev/table/sql/queries.zh.md [hunk 10]
```diff
@@ -1116,68 +1111,68 @@ The start and end timestamps of group windows as well as time attributes can be
...
+      <p>结果属性是一个可用于后续需要基于时间的操作的<a href="{{ site.baseurl }}/zh/dev/table/streaming/time_attributes.html">行时间属性</a>，比如<a href="#joins">基于时间窗口的 join </a> 以及 <a href="#aggregations">分组窗口或分组窗口上的聚合</a>。</p></td>
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
      <p>返回的是一个可用于后续需要基于时间的操作的<a href="{{ site.baseurl }}/zh/dev/table/streaming/time_attributes.html">时间属性（rowtime attribute）</a>，比如<a href="#joins">基于时间窗口的 join </a> 以及 <a href="#aggregations">分组窗口或分组窗口上的聚合</a>。</p></td>
```

File: docs/dev/table/sql/queries.zh.md [hunk 11]
```diff
@@ -587,61 +587,61 @@ WHERE o.id = s.orderId AND
...
 
-        <p><b>Note:</b> Currently, only literal <code>TRUE</code> is supported as predicate for a left outer join against a lateral table.</p>
+        <p><b>注意：</b> 当前仅支持文本常量 <code>TRUE</code> 作为针对横向表的左外部联接的谓词。</p>
       </td>
     </tr>
     <tr>
       <td>
-        <strong>Join with Temporal Table Function</strong><br>
-        <span class="label label-primary">Streaming</span>
+        <strong>Join Temporal Tables 函数</strong><br>
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
        <strong>Join Temporal Table Function</strong><br>
```

这段还是不翻译吧，描述中的也对应修改一下。

File: docs/dev/table/sql/queries.zh.md [hunk 12]
```diff
@@ -587,61 +587,61 @@ WHERE o.id = s.orderId AND
...
+        <span class="label label-primary">流处理</span>
       </td>
       <td>
-        <p><a href="{{ site.baseurl }}/dev/table/streaming/temporal_tables.html">Temporal tables</a> are tables that track changes over time.</p>
-        <p>A <a href="{{ site.baseurl }}/dev/table/streaming/temporal_tables.html#temporal-table-functions">Temporal table function</a> provides access to the state of a temporal table at a specific point in time.
-        The syntax to join a table with a temporal table function is the same as in <i>Join with Table Function</i>.</p>
+        <p><a href="{{ site.baseurl }}/zh/dev/table/streaming/temporal_tables.html">Temporal Tables</a> 是跟随时间变化而变化的表。</p>
+        <p><a href="{{ site.baseurl }}/zh/dev/table/streaming/temporal_tables.html#temporal-table-functions"> Temporal Tables 函数（ Temporal table function ）</a> 提供访问 Temporal Tables 在某一时间点的状态的能力。
```
Comment (wuchong) [inline docs/dev/table/sql/queries.zh.md]:
```suggestion
        <p><a href="{{ site.baseurl }}/zh/dev/table/streaming/temporal_tables.html#temporal-table-functions"> Temporal Table Function</a> 提供访问 Temporal Tables 在某一时间点的状态的能力。
```

**Comment by wuchong** [comment]
Hi @ousheobin , do you have time to update this PR?

**Comment by ousheobin** [comment]
Hi @wuchong . Thanks for your review.  Because I have some personal matters yesterday and I will help to update the content base on your comments and translate the corresponding content in #10732 by this week. Please let me know if you have any concern on it.

**Comment by wuchong** [comment]
Thanks @ousheobin . Please also translate the corresponding content in https://github.com/apache/flink/commit/660901699f9174d80ee40616f3efd493e1cee324#diff-32af0e13b8a200b0e205ed866c49670f.

**Comment by ousheobin** [comment]
@wuchong Sure, will do it asap.

**Comment by wuchong** [comment]
Hi @ousheobin thanks for the updating, the chagnes looks good to me in general. I will make some minor change and merge it. 

Btw, please use `git rebase` instead of `git merge`, we should always avoid merge commits in the git history.

**Pull request closed without merging**